tests/integration-tests/tests/update/test_update.py

```diff
@@ -749,3 +761,295 @@ def _test_update_queue_strategy_with_running_job(
+        bucket_name,
+    )
+
+    # Create cluster with initial configuration
+    init_config_file = pcluster_config_reader()
+    cluster = clusters_factory(init_config_file)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = scheduler_commands_factory(remote_command_executor)
+    cluster_nodes = scheduler_commands.get_compute_nodes()
+    queue1_nodes = [node for node in cluster_nodes if "queue1" in node]
+
+    # submit a job to queue1
+    queue1_job_id = scheduler_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "command": "sleep 3000",
+            "nodes": 1,
+            "partition": "queue1",
+        },
+    )
+
+    # Wait for the job to run
+    scheduler_commands.wait_job_running(queue1_job_id)
+
+    # update cluster to add ebs, efs, fsx with drain strategy
+    update_cluster_config = pcluster_config_reader(
+        config_file="pcluster.config.update_drain.yaml",
+        volume_id=ebs_volume_id,
+        existing_ebs_mount_dir=existing_ebs_mount_dir,
+        existing_efs_mount_dir=existing_efs_mount_dir,
+        fsx_lustre_mount_dir=fsx_lustre_mount_dir,
+        fsx_ontap_mount_dir=fsx_ontap_mount_dir,
+        fsx_open_zfs_mount_dir=fsx_open_zfs_mount_dir,
+        ebs_volume_id=ebs_volume_id,
+        existing_efs_id=existing_efs_id,
+        existing_fsx_lustre_fs_id=existing_fsx_lustre_fs_id,
+        fsx_ontap_volume_id=fsx_ontap_volume_id,
+        fsx_open_zfs_volume_id=fsx_open_zfs_volume_id,
+        bucket_name=bucket_name,
+        new_ebs_mount_dir=new_ebs_mount_dir,
+        new_raid_mount_dir=new_raid_mount_dir,
+        new_lustre_mount_dir=new_lustre_mount_dir,
+        new_efs_mount_dir=new_efs_mount_dir,
+    )
+
+    cluster.update(str(update_cluster_config))
+
+    # check chef client log contains expected log
+    assert_errors_in_logs(
+        remote_command_executor,
+        ["/var/log/chef-client.log"],
+        [
+            "All queues will be updated in order to update shared storages",
+        ],
+    )
+
+    scheduler_commands.assert_job_state(queue1_job_id, "RUNNING")
+    # assert all nodes are drain
+    assert_compute_node_states(
+        scheduler_commands, cluster_nodes, expected_states=["draining", "draining!", "drained*", "drained"]
+    )
+
+    # check file systems are visible on the head node just after the update
+    _test_shared_storages_mount_on_headnode(
+        remote_command_executor,
+        cluster,
+        region,
+        bucket_name,
+        scheduler_commands_factory,
+        ebs_mount_dirs,
+        new_raid_mount_dir,
+        efs_mount_dirs,
+        fsx_mount_dirs,
+    )
+
+    # check newly mounted file systems are not visible on compute nodes that are running jobs
+    for node_name in queue1_nodes:
+        _test_directory_not_mounted(remote_command_executor, all_mount_dirs, node_type="compute", node_name=node_name)
+
+    scheduler_commands.cancel_job(queue1_job_id)
```

Comment:
Shouldn't we wait some time between job cancellation and the verification that the expected shared storages are mounted on compute nodes?